wit/types.wit

```diff
@@ -0,0 +1,29 @@
+@unstable(feature = tls)
+interface types {
+    @unstable(feature = tls)
+    use wasi:io/streams@0.2.3.{input-stream, output-stream};
+    @unstable(feature = tls)
+    use wasi:io/poll@0.2.3.{pollable};
+
+    @unstable(feature = tls)
+    resource client-handshake {
+        @unstable(feature = tls)
+        constructor(server-name: string, input: input-stream, output: output-stream);
+
+        @unstable(feature = tls)
+        finish: static func(this: client-handshake) -> future-client-streams;
+    }
+
+    @unstable(feature = tls)
+    resource client-connection {
```

Comment:
one thing we will need eventually for dotnet is the `server-identity: func() -> option<public-identity>;`. After accepting the connection dotnet SSL client  runs through a second round of server certificate validation.